config/initializers/mime_types.rb

```diff
@@ -2,3 +2,9 @@
 
 # Add new mime types for use in respond_to blocks:
 # Mime::Type.register "text/richtext", :rtf
+
+jsonapi_mime = Mime::Type.register "application/vnd.api+json", :json
+
+ActionDispatch::Request.parameter_parsers[jsonapi_mime.symbol] = lambda do |body|
+  ActiveSupport::JSON.decode body
```

Comment:
Does ActiveSupport rescue invalid json decodes with something reasonable? If not, you should.

Reply:
Nope, it fails horribly, this is something I ran into when it starts parsing the query string as a body on `HEAD` requests (which is another wtf for me right now)

Reply:
I know of a company that has a line of code that looks something like this:

```ruby
begin
  JSON.decode body
rescue
  { json: :decode_failure }
end
```

Reply:
:|